internal/kafka/internal/config/kafka_quota.go

```diff
@@ -3,13 +3,15 @@ package config
 import "github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/api"
 
 type KafkaQuotaConfig struct {
-	Type                   string
-	AllowDeveloperInstance bool
+	Type                    string
+	AllowDeveloperInstance  bool
+	MaxAllowedTrialInstance int
 }
 
 func NewKafkaQuotaConfig() *KafkaQuotaConfig {
 	return &KafkaQuotaConfig{
-		Type:                   api.QuotaManagementListQuotaType.String(),
-		AllowDeveloperInstance: true,
+		Type:                    api.QuotaManagementListQuotaType.String(),
+		AllowDeveloperInstance:  true,
+		MaxAllowedTrialInstance: 1,
```

Comment:
I wonder if now `AllowDeveloperInstance` and `MaxAllowedTrialInstance` are now redundant?
Wouldn't a `MaxAllowedTrialInstance` equaling 0 the same as AllowDeveloperInstance = false?

What do you think @ziccardi @machi1990 ?

Reply:
Good point. I think merging this PR will obsolete `AllowDeveloperInstance`

Reply:
+1.

Reply:
should I remove AllowDeveloperInstance flag in this PR ?

Reply:
No harm in keeping it.

I'd suggest we contain this PR in just making the max developer instance configurable and open a follow up JIRA to remove `AllowDeveloperInstance` flag.

Reply:
I agree, but in this PR itself the `MaxAllowedTrialInstance` attribute should be renamed to `MaxAllowedDeveloperInstances`, and its corresponding CLI flags too